### ISSUE TITLE
"ERROR: 'remote' was requested for default strategies but no strategy with that identifier was registered" when using `--remote_local_fallback` and remote executor is down

### ISSUE THREAD

**Issue by fwingerter-Ocient** [bazelbuild/bazel#13487]
### Description of the problem:

Even with `--remote_local_fallback`, builds fail when the remote executor is not reachable.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```
$ git clone https://github.com/bazelbuild/examples/ bazel-examples
$ cd bazel-examples/cpp-tutorial/stage1
$ bazel build //... --remote_local_fallback --remote_executor=asdkfasdf:6890
Starting local Bazel server and connecting to it...
INFO: Invocation ID: fe0d20a4-cc89-4867-bce1-8cf6331fb814
WARNING: Failed to query remote execution capabilities: UNAVAILABLE: Unable to resolve host asdkfasdf
INFO: Analyzed target //main:hello-world (15 packages loaded, 57 targets configured).
INFO: Found 1 target...
ERROR: 'remote' was requested for default strategies but no strategy with that identifier was registered. Valid values are: [processwrapper-sandbox, standalone, worker, sandboxed, local]
INFO: Elapsed time: 5.410s
INFO: 0 processes.
FAILED: Build did NOT complete successfully
```

The `WARNING`s are expected, but the `ERROR` is not. The `ERROR` can be prevented by passing `--spawn_strategy=local`, but this prevents using the remote executor if it is up.

### What operating system are you running Bazel on?

Ubuntu 16.04.6 LTS (I know it's ancient, but I don't believe this problem is distro-related)

### What's the output of `bazel info release`?

release 4.1.0rc1

### What's the output of `git remote get-url origin ; git rev-parse master ; git rev-parse HEAD` ?

```
https://github.com/bazelbuild/examples/
b1c5b85050fa5f199f40183c84a531e458dcf8aa
b1c5b85050fa5f199f40183c84a531e458dcf8aa
```

###  Have you found anything relevant by searching the web?

It seems that this behavior is not intended according to this commit: https://github.com/bazelbuild/bazel/commit/39512a43cabf8c9a5d1c630e32dfa33d2b3fa64d

**Comment by coeuvre** [comment]
39512a4 was for remote cache. The underlying problem for this error is #13340.

**Comment by fwingerter-Ocient** [comment]
The difference between this and #13340 is that I'm not explicitly setting the list of strategies at all here, and the error complains about the "default strategies" set in Bazel's implementation (as far as I can tell).

I'm happy with any changes that fix this use-case, of course. Thanks for looking into this.

**Comment by coeuvre** [comment]
Should be fixed by #13490.
